prometheus/http.go

```diff
@@ -15,14 +15,169 @@ package prometheus
 
 import (
 	"bufio"
+	"compress/gzip"
+	"fmt"
 	"io"
+	"log"
 	"net"
 	"net/http"
 	"strconv"
 	"strings"
 	"time"
+
+	"github.com/prometheus/common/expfmt"
+)
+
+const (
+	contentTypeHeader     = "Content-Type"
+	contentLengthHeader   = "Content-Length"
+	contentEncodingHeader = "Content-Encoding"
+	acceptEncodingHeader  = "Accept-Encoding"
+)
+
+// Handler returns an HTTP handler for the DefaultRegistry. It is
+// already instrumented with InstrumentHandler (using "prometheus" as handler
+// name).
+//
+// Please note the issues described in the doc comment of InstrumentHandler. You
+// might want to consider using UninstrumentedHandler instead.
+//
+// The returned Handler is using the same HandlerOpts as the Handler returned by
+// UninstrumentedHandler. See there for details.
+func Handler() http.Handler {
```

Comment:
> It is reducing the mental load, as when e.g. I'm looking for functions related to the Registry they're right there as methods rather than spread around the package.

Yes, everything else put aside, grouping can be helpful in that way. But it's a secondary concern. I'd prefer packages over that, where a package has a small enough surface to not run into the issue in the first place. I pack things in structs primarily not for grouping but for other reasons, because it helps the code using the functions, or - as a specific Go concern - because I want to implement interfaces.

To not devolve that into a fundamental discussion about software design: We are discussing two concrete cases here, where the above matters: The push related functions and the registry related functions. My point for the push related functions is that having a struct complicates pushing by 33 to 50%. That has priority over grouping the functions. Also, we can use a package for that, in that particular case. For the registry, the grouping matters more because of the many legacy functions we have. I hope we'll be able to cut this down, currently working on it with scrutiny. The core question with the registry is if we want a registry `interface` or a registry `struct`. This is an open question (and yes, I know you believe it should be a `struct`. Still I have to think about it. Opinions of others are welcome, BTW, to speed up my thought process). If it's an interface, there is no way we would load it up with methods. If it is a struct, our options have to be weighted against each other. But let's first get the decision about the interface done. Otherwise, our endless discussion is merely academic (not a bad thing per se, but we have to get forward with this PR). 

> Classes are a Go way to group functions related to a specific type.

Go doesn't even have classes. (OK, it's a nit, but it speaks volumes about the Go mindset.)

Also, as a data point, note how https://golang.org/pkg/strings/ is a package, while the equivalent are all string methods in Java.

In different news, I see it as the perfect way to unidiomatic Go code to treat Go as similar to Java. But yeah, let's not have a fundamental programming paradigm discussion here. Let's just decide about the issues at hand. See above.